assets/js/googlesitekit/components-gm3/Checkbox.js

```diff
@@ -47,47 +59,33 @@ export default function Checkbox( {
 } ) {
 	const ref = useRef( null );
 
-	useEffect( () => {
+	const updateCheckedState = useCallback( () => {
 		const { current } = ref;
 
 		if ( ! current ) {
 			return;
 		}
 
-		function updateCheckedState() {
-			current.checked = checked;
-
-			// The checked property doesn't get reflected to the inner input element checked state,
-			// so we need to set it manually.
-			const innerInput = current.shadowRoot.querySelector( 'input' );
-			if ( innerInput ) {
-				innerInput.checked = checked;
-			}
-		}
-
-		function handleChange( event ) {
-			onChange?.( event );
+		current.checked = checked;
 
-			// Restore current checked state for controlled component behaviour.
-			updateCheckedState();
+		// The checked property doesn't get reflected to the inner input element checked state,
+		// so we need to set it manually.
+		const innerInput = current.shadowRoot.querySelector( 'input' );
+		if ( innerInput ) {
+			innerInput.checked = checked;
 		}
```

Comment:
This should no longer be needed with `@lit-labs/react` wrapped components as the passed in props are set directly as property on the element, rather than as attributes as it would when custom elements are used directly in React.